Runtimes/Core/cmake/modules/AvailabilityMacros.cmake

```diff
@@ -5,4 +5,22 @@ file(STRINGS "${CMAKE_CURRENT_BINARY_DIR}/availability-macros.def" availability_
 list(FILTER availability_defs EXCLUDE REGEX "^\\s*(#.*)?$")
 foreach(def ${availability_defs})
   add_compile_options("$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xfrontend -define-availability -Xfrontend \"${def}\">")
+
+  if("${def}" MATCHES "SwiftStdlib .*")
+    # For each SwiftStdlib x.y, also define SwiftStdlibTargetOS x.y, which,
+    # will expand to the current `-target` platform if the macro defines a
+    # newer platform as its availability.
+    #
+    # There is a setting, SwiftCore_ENABLE_STRICT_AVAILABILITY, which if set
+    # ON will cause us to use the "proper" availability instead.
+    string(REPLACE "SwiftStdlib" "SwiftStdlibCurrentOS" current "${def}")
+    if(NOT SwiftCore_ENABLE_STRICT_AVAILABILITY AND SwiftCore_SWIFT_AVAILABILITY_PLATFORM)
+      string(REGEX MATCH "${SwiftCore_SWIFT_AVAILABILITY_PLATFORM} ([0-9]+(\.[0-9]+)+)" platform_version "${def}")
```

Comment:
AvailabilityMacros.cmake is loaded before platform info, so `SwiftCore_SWIFT_AVAILABILITY_PLATFORM` won't be set by that here.